### ISSUE TITLE
Failed to execute mvn test on aarch64

### ISSUE THREAD

**Issue by simon28li** [netty/netty#12948]
### Expected behavior
Hope that the execution of `mvn test` will be successful on aarch64.

### Actual behavior
I saw the error 
/tmp/com_aayushatharva_brotli4j_2094286067179980/libbrotli.so: / lib64/libm.so.6: version GLIBC_2.29 not found (required by /tmp/com_aayushatharva_brotli4j_2094286067179980/libbrotli.so),
 whether it was caused by a lower version of libm, but on x86 machines, I checked that libm also did not contain GLIBC_2.29 but passed the test normally.

The error is reported after the test is executed as follows.

```
[INFO] Running io.netty.handler.codec.compression.BrotliDecoderTest
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.002 s <<< FAILURE! - in io.netty.handler.codec.compression.BrotliDecoderTest
[ERROR] io.netty.handler.codec.compression.BrotliDecoderTest  Time elapsed: 0.002 s  <<< ERROR!
java.lang.ExceptionInInitializerError
        at io.netty.handler.codec.compression.BrotliDecoderTest.setUp(BrotliDecoderTest.java:56)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
        at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
        at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
        at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:156)
        at org.junit.jupiter.engine.extension.TimeoutExtension.interceptLifecycleMethod(TimeoutExtension.java:128)
        at org.junit.jupiter.engine.extension.TimeoutExtension.interceptBeforeAllMethod(TimeoutExtension.java:70)
        at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(InterceptingExecutableInvoker.java:103)
        at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.lambda$invoke$0(InterceptingExecutableInvoker.java:93)
        at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
        at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
        at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
        at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
        at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:92)
        at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:86)
        at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$invokeBeforeAllMethods$13(ClassBasedTestDescriptor.java:411)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeBeforeAllMethods(ClassBasedTestDescriptor.java:409)
        at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:215)
        at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:84)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:148)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
        at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
        at java.util.ArrayList.forEach(ArrayList.java:1257)
        at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
        at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
        at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
        at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
        at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
        at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:220)
        at org.junit.platform.launcher.core.DefaultLauncher.lambda$execute$6(DefaultLauncher.java:188)
        at org.junit.platform.launcher.core.DefaultLauncher.withInterceptedStreams(DefaultLauncher.java:202)
        at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:181)
        at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:128)
        at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:150)
        at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:124)
        at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
        at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
        at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
        at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
Caused by: java.lang.UnsatisfiedLinkError: Failed to load Brotli native library
        at com.aayushatharva.brotli4j.Brotli4jLoader.ensureAvailability(Brotli4jLoader.java:79)
        at io.netty.handler.codec.compression.Brotli.ensureAvailability(Brotli.java:71)
        at io.netty.handler.codec.compression.BrotliDecoderTest.setUp(BrotliDecoderTest.java:48)
        ... 55 more
Caused by: java.lang.UnsatisfiedLinkError: /tmp/com_aayushatharva_brotli4j_2094286067179980/libbrotli.so: /lib64/libm.so.6: version `GLIBC_2.29' not found (required by /tmp/com_aayushatharva_brotli4j_2094286067179980/libbrotli.so)
        at java.lang.ClassLoader$NativeLibrary.load(Native Method)
        at java.lang.ClassLoader.loadLibrary0(ClassLoader.java:1941)
        at java.lang.ClassLoader.loadLibrary(ClassLoader.java:1824)
        at java.lang.Runtime.load0(Runtime.java:809)
        at java.lang.System.load(System.java:1086)
        at com.aayushatharva.brotli4j.Brotli4jLoader.<clinit>(Brotli4jLoader.java:56)
        at io.netty.handler.codec.compression.Brotli.<clinit>(Brotli.java:46)
        at io.netty.handler.codec.compression.BrotliEncoderTest.setUp(BrotliEncoderTest.java:32)
        ... 55 more

[INFO] Running io.netty.handler.codec.ReplayingDecoderByteBufTest
```

### Steps to reproduce
mvn test

### Netty version
4.1.79

### JVM version (e.g. `java -version`)
"1.8.0_191"

### OS version (e.g. `uname -a`)
Linux  4.18.0-383.el8.aarch64 #1 SMP Wed Apr 20 15:39:57 UTC 2022 aarch64 aarch64 aarch64 GNU/Linux /CentOS 8

**Comment by chrisvest** [comment]
@hyperxpro what's your take?

**Comment by hyperxpro** [comment]
Brotli4j aarch64 is compressed for Ubuntu 20.04 GLIBC. I will add backward compatibility like x86_64 binary by next week probably.

**Comment by simon28li** [comment]
> Brotli4j aarch64 is compressed for Ubuntu 20.04 GLIBC. I will add backward compatibility like x86_64 binary by next week probably.

Does this mean that other aarch64 systems other than ubuntu, such as centos and suse, do not support testing?

**Comment by simon28li** [comment]
> > Brotli4j aarch64 is compressed for Ubuntu 20.04 GLIBC. I will add backward compatibility like x86_64 binary by next week probably.
> 
> Does this mean that other aarch64 systems other than ubuntu, such as centos and suse, do not support testing?

I repeated the consistent error in the ubuntu 18.04.5 test

**Comment by hyperxpro** [comment]
Nope. It's just compiled with a newer version of GLIBC. Just have the GLIBC_2.29 or above and you're good to go

**Comment by hyperxpro** [comment]
> > > Brotli4j aarch64 is compressed for Ubuntu 20.04 GLIBC. I will add backward compatibility like x86_64 binary by next week probably.
> > 
> > 
> > Does this mean that other aarch64 systems other than ubuntu, such as centos and suse, do not support testing?
> 
> I repeated the consistent error in the ubuntu 18.04.5 test

That's because Ubuntu 18.04 is below GLIBC_2.29.

**Comment by normanmaurer** [comment]
@hyperxpro did you fix this ?

**Comment by simon28li** [comment]
> Yes, aarch64 support is now available in 1.9.0 release. We can close this.
Does version 1.9.0 correspond to the 4.x.x version?

**Comment by hyperxpro** [comment]
Brotli4j 1.9.0 I'm talking about.
